tm2py/components/network/highway/highway_assign.py

```diff
@@ -164,9 +164,19 @@ def run(self):
                     # (2) run_dynamic_toll = True, warmstart = True
                     # run maximum 5 times of dynamic tolling
                     # break out the loop if no valuetoll need to be updated
+
+                    # pre-check if there are any valuetoll links
+                    valuetoll_start_tollbooth_code = self.config.tolls.valuetoll_start_tollbooth_code
+                    valuetoll_links_exist = False
+                    network = scenario.get_network()
+                    for link in network.links():
+                        if link["@tollbooth"] >= valuetoll_start_tollbooth_code and link["@lanes"] > 0:
+                            valuetoll_links_exist = True
```

Comment:
Is the logic: if there is no valuetoll links (`valuetoll_links_exist == False`), then run assignment without dynamic tolling? The way it is coded right now still enters the dynamic tolling for loop below which is unnecessary. Operationally the current code works, but it can benefit from some clean up and re-org. To me, `valuetoll_links_exist` should not be part of this else condition starting line 162. Let me know if I am misunderstanding anything.

Reply:
adjusted in bde416d